src/Altinn.App.Core/Helpers/ObjectUtils.cs

```diff
@@ -137,14 +138,31 @@ public static void PrepareModelForXmlStorage(object model, int depth = 64)
                 SetToDefaultIfShouldSerializeFalse(model, prop, methodInfos);
 
                 // Set string properties with [XmlText] attribute to null if they are empty or whitespace
-                if (
-                    value is string s
-                    && string.IsNullOrWhiteSpace(s)
-                    && prop.GetCustomAttribute<XmlTextAttribute>() is not null
-                )
+                if (value is string s)
                 {
-                    // Ensure empty strings are set to null
-                    prop.SetValue(model, null);
+                    if (string.IsNullOrWhiteSpace(s))
+                    {
+                        if (prop.GetCustomAttribute<XmlTextAttribute>() is not null)
```

Comment:
I don't think the nesting here makes sense. We should continue matching regex to remove FORM-FEED ++ when the string is whitespace and not `[XmlText]`.